gormigrate.go

```diff
@@ -394,6 +411,31 @@ func (g *Gormigrate) canInitializeSchema() (bool, error) {
 	return count == 0, err
 }
 
+func (g *Gormigrate) unknownMigrationsHaveHappened() (bool, error) {
+	sql := fmt.Sprintf("SELECT %s FROM %s", g.options.IDColumnName, g.options.TableName)
+	rows, err := g.tx.Raw(sql).Rows()
+	if err != nil {
+		return false, err
+	}
+	defer rows.Close()
+
+	validIDSet := make(map[string]bool, len(g.migrations)+1)
```

Comment:
https://dave.cheney.net/2014/03/25/the-empty-struct

```suggestion
	validIDSet := make(map[string]struct{}, len(g.migrations)+1)
```